homeassistant/components/nws/diagnostics.py

```diff
@@ -0,0 +1,32 @@
+"""Diagnostics support for NWS."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from pynws import SimpleNWS
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_STATION, DOMAIN
+
+CONFIG_TO_REDACT = {CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_STATION}
+OBSERVATION_TO_REDACT = {"station"}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    nws_data: SimpleNWS = hass.data[DOMAIN][config_entry.entry_id].api
```

Comment:
Just fyi as I saw it, this could make use of `config_entry.runtime_data` (https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry). But this is for a new PR.